open_spiel/python/examples/leduc_nfsp.py

```diff
@@ -67,120 +71,330 @@
                    "Starting exploration parameter.")
 flags.DEFINE_float("epsilon_end", 0.001,
                    "Final exploration parameter.")
-flags.DEFINE_string("evaluation_metric", "nash_conv",
-                    "Choose from 'exploitability', 'nash_conv'.")
+flags.DEFINE_enum("evaluation_metric", "nash_conv", ["exploitability", "nash_conv", "avg_return"],
+                  "Choose from 'exploitability', 'nash_conv', 'avg_return'.")
+flags.DEFINE_integer("evaluation_opponent_pool_size", 5,
+                     "Only affects the 'avg_return' evaluation metric. How many past checkpoints to use as the pool of opponents.")
+flags.DEFINE_enum("evaluation_opponent_pool", "stratified", ["recent", "random", "stratified"],
+                  "Only affects the 'avg_return' evaluation metric.  Determines which how to sample the pool of past opponents to use when evaluating average returns.")
+flags.DEFINE_enum("evaluation_opponent_sampling", "independent", ["independent", "correlated", "perturbed"],
+                  "Only affects the' avg_return' evaluation metric.  Determines how to sample rosters of opponents from the pool of possible opponents."
+                  "Options are 'indpendent' to sample each player independently, 'correlated' to copy entire rosters from a previous episode, "
+                  "and 'perturbed' to use the current episode's roster with a single opponent copied from a previous episode.")
+flags.DEFINE_integer("evaluation_num_samples", 1000,
+                     "Only affects the 'avg_return' evaluation metric.  How many episodes of play to sample for calculating the average return.")
+
 flags.DEFINE_bool("use_checkpoints", True, "Save/load neural network weights.")
 flags.DEFINE_string("checkpoint_dir", "/tmp/nfsp_test",
-                    "Directory to save/load the agent.")
-
+"Directory to save/load the agent.")
 
 class NFSPPolicies(policy.Policy):
-  """Joint policy to be evaluated."""
-
-  def __init__(self, env, nfsp_policies, mode):
-    game = env.game
-    player_ids = list(range(FLAGS.num_players))
-    super(NFSPPolicies, self).__init__(game, player_ids)
-    self._policies = nfsp_policies
-    self._mode = mode
-    self._obs = {
-        "info_state": [None] * FLAGS.num_players,
-        "legal_actions": [None] * FLAGS.num_players
-    }
+    """Joint policy to be evaluated."""
 
-  def action_probabilities(self, state, player_id=None):
-    cur_player = state.current_player()
-    legal_actions = state.legal_actions(cur_player)
+    def __init__(self, env, nfsp_policies, mode):
```

Comment:
Why is the constructior inside action probabilites? I think this must have been accidental? Can you move it back?